### ISSUE TITLE
networkd: WireGuardPeer sections not picked up from drop-in directory

### ISSUE THREAD

**Issue by retroflexyz** [systemd/systemd#12070]
**systemd version the issue has been seen with**
> systemd 239
+PAM -AUDIT -SELINUX +IMA -APPARMOR +SMACK -SYSVINIT +UTMP -LIBCRYPTSETUP +GCRYPT -GNUTLS +ACL -XZ +LZ4 +SECCOMP +BLKID -ELFUTILS +KMOD -IDN2 -IDN +PCRE2 default-hierarchy=hybrid

**Used distribution**
 > gentoo

**Expected behaviour you didn't see**
 > `[WireGuardPeer]` sections defined in drop-in folder creates peer entries for WireGuard

**Unexpected behaviour you saw**
 > `[WireGuardPeer]` definitions in drop-in folder appear to have no effect.

**Steps to reproduce the problem**
Attempting to use `networkd` to configure WireGuard "server", with configuration for peers in `*.netdev.d/`:

network/30-wg0.netdev:
```
[NetDev]
Name=wg0
Kind=wireguard

[WireGuard]
PrivateKey=...
ListenPort=51820
```

network/30-wg0.netdev.d/device.conf:
```
[WireGuardPeer]
PublicKey=ABCD...
AllowedIPs=...
```

After `networkd` starts, it is expected that `wg show` would include a peer with the key `ABCD...`. However, no peer entries are created.

The same peer configuration works as expected when merged into `30-wg0.netdev`.

**Comment by yuwata** [comment]
I hope this issue is fixed recently by #11861. If possible, please try the current git master.

**Comment by yuwata** [comment]
I've added a test for the issue in #12157, and confirmed that this issue is already fixed by #11861.
